### PR TITLE
fix!: rename scheduled event parameter `channel_id` to `channel`, fix validation

### PR DESCRIPTION
**Pull request by shiftinv**
## Summary

Changes:
- Rename the `channel_id` parameter for scheduled event create/edit to `channel` to improve consistency with the rest of the code
- Add overloads to `Guild.create_scheduled_event` and `GuildScheduledEvent.edit`
- Make `scheduled_end_time` parameter optional
- Remove (slightly broken) parameter validation, as some parameters were being required even though not providing them would've been fine, particularly when editing an event's type

## Checklist

- [x] If code changes were made, then they have been tested
    - [x] I have updated the documentation to reflect the changes
    - [x] I have formatted the code properly by running `task lint`
    - [x] I have type-checked the code by running `task pyright`
- [x] This PR fixes an issue
- [ ] This PR adds something new (e.g. new method or parameters)
- [x] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

**Comment by shiftinv** [comment]
Updated this branch to also add overloads and improve/fix some of the validation related to editing scheduled events, still somewhat wip.
Instead of only deprecating `channel_id` as originally planned, it's now removed completely.

~~it's not scope creep if the original implementation got scrapped entirely, right?~~

File: disnake/guild.py [hunk 1]
```diff
@@ -2124,8 +2159,8 @@ async def create_scheduled_event(
             .. versionchanged:: 2.5
                 Now accepts various resource types in addition to :class:`bytes`.
 
-        channel_id: :class:`int`
-            The channel ID in which the guild scheduled event will be hosted.
+        channel: :class:`.abc.Snowflake`
+            The channel in which the guild scheduled event will be hosted.
```
Comment (onerandomusername) [inline disnake/guild.py:2163]: versionadded

File: disnake/guild_scheduled_event.py [hunk 2]
```diff
@@ -300,14 +360,14 @@ async def edit(
             .. versionchanged:: 2.5
                 Now accepts various resource types in addition to :class:`bytes`.
 
-        channel_id: Optional[:class:`int`]
-            The channel ID in which the guild scheduled event will be hosted.
+        channel: Optional[:class:`.abc.Snowflake`]
+            The channel in which the guild scheduled event will be hosted.
```
Comment (onerandomusername) [inline disnake/guild_scheduled_event.py:364]: versionadded

File: disnake/guild_scheduled_event.py [hunk 3]
```diff
@@ -282,7 +343,6 @@ async def edit(
```
Comment (onerandomusername) [inline disnake/guild_scheduled_event.py:343]: versionchanged

File: disnake/guild.py [hunk 4]
```diff
@@ -2104,7 +2139,7 @@ async def create_scheduled_event(
```
Comment (onerandomusername) [inline disnake/guild.py:2139]: versionchanged

File: disnake/guild.py (hunk 1 above)
Comment (shiftinv) [reply disnake/guild.py:2166]: d4a8082e76bd518efa259b35d33d4f46b95a1151

File: disnake/guild_scheduled_event.py [hunk 5]
```diff
@@ -261,16 +261,77 @@ async def delete(self) -> None:
...
+    async def edit(
+        self,
+        *,
+        entity_type: Literal[GuildScheduledEventEntityType.external],
+        name: str = ...,
+        description: Optional[str] = ...,
+        image: Optional[AssetBytes] = ...,
+        privacy_level: GuildScheduledEventPrivacyLevel = ...,
+        scheduled_start_time: datetime = ...,
+        scheduled_end_time: Optional[datetime] = ...,
```
Comment (onerandomusername) [inline disnake/guild_scheduled_event.py:297]: is this supposed to be optional for external events?

Comment (shiftinv) [reply disnake/guild_scheduled_event.py]: Nope, good catch. Fixed in 9f9b48793e089dcd5972a2378503c5a708289407

**Review by onerandomusername — APPROVED** [review]

**Pull request merged**